src/transformers/generation/utils.py

```diff
@@ -351,6 +359,43 @@ def prepare_inputs_for_generation(self, *args, **kwargs):
             "A model class needs to define a `prepare_inputs_for_generation` method in order to use `.generate()`."
         )
 
+    def set_cache_implementation(self, cache_implementation: Union[CacheImplementation, str], **kwargs):
```

Comment:
I'd rather have this function as `set_cache`, with `past_key_values` as an optional parameter. But this is a tougher discussion: I've opened the discussion on our internal slack [here](https://huggingface.slack.com/archives/C060NPVEW57/p1708006459508009) 🤗